### PR TITLE
kic: limit SSH auth to only private key

### PR DESCRIPTION
**Pull request by medyagh**

### CODE REVIEW

**Comment by medyagh** [comment]
build-base-image

**Comment by medyagh** [comment]
ok-to-build-image

File: deploy/kicbase/Dockerfile [hunk 1]
```diff
@@ -227,9 +227,7 @@ RUN systemctl enable docker.service
 # making SSH work for docker container
 # based on https://github.com/rastasheep/ubuntu-sshd/blob/master/18.04/Dockerfile
 RUN mkdir /var/run/sshd
-RUN echo 'root:root' |chpasswd
-RUN sed -ri 's/^#?PermitRootLogin\s+.*/PermitRootLogin yes/' /etc/ssh/sshd_config
-RUN sed -ri 's/UsePAM yes/#UsePAM yes/g' /etc/ssh/sshd_config
+RUN sed -ri 's/^#?PermitRootLogin\s+.*/PermitRootLogin no/' /etc/ssh/sshd_config
```
Comment (afbjorklund) [inline deploy/kicbase/Dockerfile:230]: You can remove this sshd_config line too, and leave the sshd default: `#PermitRootLogin prohibit-password`

This allows user to talk with their containerd and other niche use cases, while still prohibiting root password.

https://minikube.sigs.k8s.io/docs/handbook/pushing/#6-pushing-directly-to-in-cluster-containerd-buildkitd

The line with "UsePAM" is unrelated to this change, so probably leave it - or remove it in a separate commit ?

**Review by afbjorklund — CHANGES_REQUESTED** [review]

File: deploy/kicbase/Dockerfile (hunk 1 above)
Comment (medyagh) [reply deploy/kicbase/Dockerfile:230]: sounds good ! thank you for pointing it out, I just updated it

**Comment by medyagh** [comment]
ok-to-build-image

**Comment by spowelljr** [comment]
ok-to-build-image

**Review by spowelljr — APPROVED** [review]

**Pull request merged**